tests/unit/splinter_shmem_test.c

```diff
@@ -0,0 +1,658 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * -----------------------------------------------------------------------------
+ * splinter_shmem_test.c --
+ *
+ *  Exercises the interfaces in SplinterDB shared memory allocation module.
```

Comment:
Very useful and interesting shmem-focussed testing.

Has been very helpful to me during shmem-dev.

Test has been greatly enhanced and expanded, especially in part-3.